util/genrest/resttools/populatefield.go

```diff
@@ -0,0 +1,140 @@
+	for idx, fieldName := range levels {
+		messageDescriptor := message.Descriptor()
+		messageFullName := messageDescriptor.FullName()
+
+		if messageDescriptor.Syntax() != protoreflect.Proto3 {
+			return fmt.Errorf("cannot process %q as it does not use proto3 syntax", messageFullName)
+		}
+
+		if len(fieldName) == 0 {
+			return fmt.Errorf("segment %d of path field %q is empty", idx, fieldPath)
+		}
+
+		// find field
+		subFields := messageDescriptor.Fields()
+		fieldDescriptor := subFields.ByName(protoreflect.Name(fieldName))
+		if fieldDescriptor == nil {
+			return fmt.Errorf("could not find %dth field (%q) in field path %q in message %q",
+				idx, fieldName, fieldPath, messageFullName)
+		}
+
+		if idx != lastLevel {
+			// non-terminal field: go to the next level
+
+			if kind := fieldDescriptor.Kind(); kind != protoreflect.MessageKind {
+				return fmt.Errorf("%dth field (%q) in field path %q in message %q is not itself a message but a %q",
+					idx, fieldName, fieldPath, messageFullName, kind)
+			}
+			message = message.Mutable(fieldDescriptor).Message()
+			continue
+		}
+
+		// terminal field
+		var (
+			protoValue protoreflect.Value
+			parseError error
+		)
+		kind := fieldDescriptor.Kind()
+		switch kind {
+		case protoreflect.MessageKind:
+			parseError = fmt.Errorf("terminal field %q of field path %q in message %q is a message type",
+				fieldName, fieldPath, messageFullName)
+
+		// reference for proto scalar types:
+		// https://developers.google.com/protocol-buffers/docs/proto3#scalar
+
+		case protoreflect.StringKind:
+			protoValue = protoreflect.ValueOfString(value)
+		case protoreflect.BytesKind:
+			protoValue = protoreflect.ValueOfBytes([]byte(value))
+
+		case protoreflect.Int32Kind, protoreflect.Sint32Kind, protoreflect.Sfixed32Kind:
+			v64, err := strconv.ParseInt(value, 10, 32)
+			parseError, protoValue = err, protoreflect.ValueOfInt32(int32(v64))
+		case protoreflect.Uint32Kind, protoreflect.Fixed32Kind:
+			v64, err := strconv.ParseUint(value, 10, 32)
+			parseError, protoValue = err, protoreflect.ValueOfUint32(uint32(v64))
+
+		case protoreflect.Int64Kind, protoreflect.Sint64Kind, protoreflect.Sfixed64Kind:
+			v64, err := strconv.ParseInt(value, 10, 64)
+			parseError, protoValue = err, protoreflect.ValueOfInt64(v64)
+		case protoreflect.Uint64Kind, protoreflect.Fixed64Kind:
+			v64, err := strconv.ParseUint(value, 10, 64)
+			parseError, protoValue = err, protoreflect.ValueOfUint64(v64)
+
+		case protoreflect.FloatKind:
+			v64, err := strconv.ParseFloat(value, 32)
+			parseError, protoValue = err, protoreflect.ValueOfFloat32(float32(v64))
+		case protoreflect.DoubleKind:
+			v64, err := strconv.ParseFloat(value, 64)
+			parseError, protoValue = err, protoreflect.ValueOfFloat64(v64)
+
+		case protoreflect.BoolKind:
+			// TODO: should we be stricter in what we accept? ParseBool accepts various
+			// representations of "true" and "false" (https://golang.org/pkg/strconv/#ParseBool)
```

Comment:
I would argue it's less confusing and frustrating for the user if we were more strict initially and became less strict as needs arose. Then again, there isn't an obvious reason to be strict in this case.

Reply:
I'm leaving it as a TODO for now and then will investigate how strictly, if at all, Google and REST conventions enforce Boolean value representations in REST calls. That said, since Showcase is meant to be used primarily by GAPICs, we could be strict here to force GAPICs to be strict in what they emit. (If somebody else uses Showcase without GAPICs, they may have issues, but we don't necessarily need to optimize for them.)